src/fermilib/ops/_fermion_operator.py

```diff
@@ -51,7 +51,7 @@ def number_operator(n_orbitals, orbital=None, coefficient=1.):
     return operator
 
 
-def normal_ordered_term(term, coefficient):
+def normal_ordered_term(term, coefficient, key_array=None):
```

Comment:
I don't think Ryan will allow the pull request either way, but key_array gives the assignment from range(n) to an arbitrary assignment. For example if you normal order 3^ 0^ by [0, 1, 2, 3] it'll leave it as is, but if you normal order by [3, 2, 1, 0] it'll reverse it to -0^ 3^. There are a lot of other examples in the test file for this pull request.